core/storage/btree.rs

```diff
@@ -290,6 +303,19 @@ impl BTreeCursor {
         &mut self,
         predicate: Option<(SeekKey<'_>, SeekOp)>,
     ) -> Result<CursorResult<(Option<u64>, Option<Record>)>> {
+        if let Some(mv_cursor) = &self.mv_cursor {
+            let mut mv_cursor = mv_cursor.borrow_mut();
+            let rowid = mv_cursor.current_row_id();
+            match rowid {
+                Some(rowid) => {
+                    let record = mv_cursor.current_row().unwrap().unwrap();
+                    let record: Record = crate::storage::sqlite3_ondisk::read_record(&record.data)?;
+                    mv_cursor.forward();
+                    return Ok(CursorResult::Ok((Some(rowid.row_id), Some(record))));
+                }
+                None => return Ok(CursorResult::Ok((None, None))),
```

Comment:
Does `mv_cursor` fill all rows in memory? I imagine we will fill the case where we need to lazy load pages to multi version store right or what am I missing?

Reply:
@pereman2 Yes it does. I opened https://github.com/tursodatabase/limbo/issues/1104 because it requires reworking the MVCC layer itself.